doc/lsst.pipe.base/creating-a-task.rst

```diff
@@ -0,0 +1,350 @@
+       - stdDevErr: uncertainty in standard deviation
+       """
+       self.log.info("Processing data ID %s" % (dataRef.dataId,))
+       if self.config.doFail:
+           raise pipeBase.TaskError("Raising TaskError by request (config.doFail=True)")
+       # Unpersist the raw exposure pointed to by the data reference
+       rawExp = dataRef.get("raw")
+       maskedImage = rawExp.getMaskedImage()
+       # Support extra debug output.
+       # -
+       import lsstDebug
+       display = lsstDebug.Info(__name__).display
+       if display:
+           frame = 1
+           mtv(rawExp, frame=frame, title="exposure")
+       # return the pipe_base Struct that is returned by self.stats.run
+       return self.stats.run(maskedImage)
+
+The statistics are actually computed by the ``stats`` subtask.
+Here is the ``run`` method for the default version of that task: ``exampleTask.ExampleSigmaClippedStatsTask.run``:
+
+.. code-block:: python
+
+   @pipeBase.timeMethod
+   def run(self, maskedImage):
+       """!Compute and return statistics for a masked image
+       @param[in] maskedImage: masked image (an lsst::afw::MaskedImage)
+       @return a pipeBase Struct containing:
+       - mean: mean of image plane
+       - meanErr: uncertainty in mean
+       - stdDev: standard deviation of image plane
+       - stdDevErr: uncertainty in standard deviation
+       """
+       statObj = afwMath.makeStatistics(maskedImage, afwMath.MEANCLIP | afwMath.STDEVCLIP | afwMath.ERRORS,
+                                        self._statsControl)
+       mean, meanErr = statObj.getResult(afwMath.MEANCLIP)
+       stdDev, stdDevErr = statObj.getResult(afwMath.STDEVCLIP)
+       self.log.info("clipped mean=%0.2f; meanErr=%0.2f; stdDev=%0.2f; stdDevErr=%0.2f" %
+                     (mean, meanErr, stdDev, stdDevErr))
+       return pipeBase.Struct(
+           mean=mean,
+           meanErr=meanErr,
+           stdDev=stdDev,
+           stdDevErr=stdDevErr,
+       )
+
+.. _creating-a-task-debug-variables:
+
+Debug variables
+===============
+
+Debug variables are variables the user may set while running your task, to enable additional debug output.
+To have your task support debug variables, have it import ``lsstDebug`` and call ``lsstDebug.Info(__name__).varname`` to get the debug variable ``varname`` specific to your task.
+If you look for a variable the user has not specified, it will have a value of `False`.
+For example, to look for a debug variable named "display":
+
+.. code-block:: python
+
+   import lsstDebug
+   display = lsstDebug.Info(__name__).display
+   if display:
+      # ...
+      pass
+
+.. FIXME create link when lsstDebug documentation is ready.
+.. See \ref baseDebug for more information about debug variables, including how to specify them while running a command-line task.
+
+.. _creating-a-task-docs:
+
+Documentation
+=============
+
+To be written.
+
+.. _creating-a-task-subtasks:
+
+Subtasks
+========
+
```

Comment:
This discussion wasn't clear (to me!) when the two methods were introduced.   I still think we need to decide how we are doing this and deprecate one version (probably `lsst.pex.config.ConfigurableField`, but I have no strong opinion)